File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int l = 0;
        int r = nums.size()-1;
        int k = r;
        vector<int> ret(nums.size());
        while(k>=0){
            if(nums[l]*nums[l] > nums[r]*nums[r]){
                ret[k--] = nums[l]*nums[l];
                l++;
            }
            else{
                ret[k--] = nums[r]*nums[r];
                r--;
            }
        }
        return ret;
    }
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l, r, k = 0 , len(nums) - 1, len(nums) - 1 
        ans = [0]*len(nums)
        while k >= 0:
            if nums[r]**2 > nums[l]**2:
                ans[k] = nums[r]**2
                k -= 1
                r -= 1
            else:
                ans[k] = nums[l]**2
                k -= 1
                l += 1

        return ans
            
            